### DISCUSSION TITLE
Custom Select Result Not Converted With Type Converter

### DISCUSSION THREAD

**Discussion by yeanpei** [simolus3/drift#2487] [Q&A] [answered]
I have a uuid column stores as blob and map to String using type converter.

`BlobColumn get uuid => blob().named("uuid").map(const UUIDConverter())();`

Type converter:
```
class UUIDConverter extends TypeConverter<String, Uint8List> {
  const UUIDConverter();

  @override
  String fromSql(Uint8List fromDb) => Uuid.unparse(fromDb);

  @override
  Uint8List toSql(String value) =>
      Uint8List.fromList(Uuid.parse((value.isEmpty ? Uuid().v4() : value)));
}
```
I use custom select to query from the table, but the result for the uuid column is not converted to String, did I miss out anything?
```
Selectable<FieldInspStatusVM> getCreationStatus() {
    final query = customSelect(
      'WITH creationStatus AS ('
      'SELECT field_insp_issue_id, MAX(status) AS c_status FROM fi_status WHERE status <= 1 GROUP BY field_insp_issue_id)'
      'SELECT b.* FROM '
      'creationStatus a INNER JOIN fi_status b ON a.field_insp_issue_id = b.field_insp_issue_id AND a.c_status = b.status',
      readsFrom: {fieldInspStatuses},
    );

    return query.asyncMap((row) => FieldInspStatusVM.fromJson(row.data));
}
```
This is **row.data** result:
![image](https://github.com/simolus3/drift/assets/51101784/d9b4e4fb-81ee-41d3-9a99-1ed56d265ff1)

**Comment by simolus3** [comment, marked as answer]
Type converters are applied by generated code. When you use a `customSelect` with raw SQL, drift doesn't know about the columns and whether a type converter should be applied to the raw map. So `row.data` will indeed just give you the direct row from the database without any interpretation by drift.

Is `FieldInspStatusVM` a table? In that case, you could use `query.asyncMap((row) => fieldInspStatuses.map(row.data, null))` to interpret the result. The `.map` method on tables is responsible for applying type converters.

**Reply by yeanpei** [reply]
The `.map` method on tables is exactly what I am looking for! Thanks for you help!!!
